lms/djangoapps/gradebook/models.py

```diff
@@ -98,23 +97,53 @@ def generate_leaderboard(cls, course_key, user_id=None, count=3, exclude_users=[
                     .order_by('-grade', 'modified')[:count]
                 # If a user_id value was provided, we need to provide some additional user-specific data to the caller
                 if user_id:
-                    user_grade = 0
-                    user_time_scored = timezone.now()
-                    try:
-                        user_queryset = StudentGradebook.objects.get(course_id__exact=course_key, user__id=user_id)
-                    except StudentGradebook.DoesNotExist:
-                        user_queryset = None
-                    if user_queryset:
-                        user_grade = user_queryset.grade
-                        user_time_scored = user_queryset.created
-                    users_above = queryset.filter(grade__gte=user_grade)\
-                        .exclude(user__id=user_id)\
-                        .exclude(grade=user_grade, modified__gt=user_time_scored)
-                    data['user_position'] = len(users_above) + 1
-                    data['user_grade'] = user_grade
+                    result = cls.get_user_position(
+                        course_key,
+                        user_id,
+                        queryset=queryset,
+                        exclude_users=exclude_users
+                    )
+                    print 'result = {}'.format(result)
+                    data.update(result)
 
         return data
 
+    @classmethod
+    def get_user_position(cls, course_key, user_id, queryset=None, exclude_users=[]):
+        """
+        Helper method to return the user's position in the leaderboard for Proficiency
```

Comment:
If this is a helper method -- which I assume means it's intended to be private -- should it be prefixed with an underscore?  I'm unsure of the conventions when it comes to class methods